src/classes/OPP_PrimaryContactRoleMerge_TEST.cls

```diff
@@ -0,0 +1,426 @@
+/*
+    Copyright (c) 2018, Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2018
+* @group Opportunity Contact Role
+* @description Test Class for the deletion of the duplicate primary OCRs.
+*/
+
+@isTest
+private class OPP_PrimaryContactRoleMerge_TEST {
+
+    /*********************************************************************************************************
+    * @description Given an Opportunity with blank Primary Contact AND the Opportunity has duplicate primary 
+    * OCRs with Role != NPSP Settings Role, when the batch runs, then an error log created, containing a link 
+    * to the Opportunity.
+    */
+    @isTest static void test_LogErrorIfNoOCRsRoleDefinedInNPSPSettings() {
+        String nonNPSPRole = 'NON_NPSP_ROLE';
+
+        deactivateTriggers();
+
+        Account account = new Account(Name = 'TEST_ACCOUNT');
```

Comment:
What do you think about moving some of the setup data to the test setup class? You could leverage class level variables to hold the account and contact records created from test setup and also some of the constant strings that repeat throughout could be added as class constants.

Reply:
Instead of querying for the results, one option would be to create the reused variables as static class level variables and then assign the results of the TestSetup there.